### ISSUE TITLE
ERROR: unable to collate and parse R files for package ‘SeuratObject’

### ISSUE THREAD

**Issue by garrett-lam** [satijalab/seurat#7965]
Currently working on an AWS EC2 instance is on Ubuntu 18.04.5 LTS (GNU/Linux 5.3.0-1032-aws x86_64). It came with R version 4.0.2 installed. I am unable to install Seurat.

I ran the following 
```
install.packages("Seurat")
```

Then received this error:
```
Error in parse(outFile) : 
  /tmp/RtmpZteZNR/R.INSTALLe537c8d04b2/SeuratObject/R/assay5.R:174:13: unexpected input
173:     expr = Layers(object = object, search = layer),
174:     error = \
                 ^
ERROR: unable to collate and parse R files for package ‘SeuratObject’
* removing ‘/home/rstudio/R/x86_64-pc-linux-gnu-library/4.0/SeuratObject’
Warning in install.packages :
  installation of package ‘SeuratObject’ had non-zero exit status
ERROR: dependency ‘SeuratObject’ is not available for package ‘Seurat’
* removing ‘/home/rstudio/R/x86_64-pc-linux-gnu-library/4.0/Seurat’
Warning in install.packages :
  installation of package ‘Seurat’ had non-zero exit status
```

I am not sure whether this error is due to my R version being 4.0.2. Please let me know if there is a way to fix this.

```
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8        LC_COLLATE=C.UTF-8    
 [5] LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8    LC_PAPER=C.UTF-8       LC_NAME=C             
 [9] LC_ADDRESS=C           LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
[1] compiler_4.0.2 tools_4.0.2   
```

**Comment by zskylarli** [comment]
Hi - have you tried installing SeuratObject first using the `install.packages("SeuratObject")` command?

**Comment by garrett-lam** [comment]
Yes, I have tried install.packages('SeuratObject'), it throws the same error shown above.

**Comment by Daisy-Chilin** [comment]
Hi, Im also having the  same issue. @garrett-lam , where you able to get this work?

This is what im using:
R version 4.0.3 (2020-10-10)
Platform: x86_64-conda-linux-gnu (64-bit)
Running under: Amazon Linux AMI 2017.03

**Comment by annammckee** [comment]
I am having the same issue, also on an AWS EC2 instance:
> sessionInfo()
R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 18.04.5 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas/libblas.so.3
LAPACK: /usr/lib/x86_64-linux-gnu/libopenblasp-r0.2.20.so

locale:
 [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8        LC_COLLATE=C.UTF-8    
 [5] LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8    LC_PAPER=C.UTF-8       LC_NAME=C             
 [9] LC_ADDRESS=C           LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] rstudioapi_0.11   knitr_1.29        munsell_0.5.0     colorspace_1.4-1  lattice_0.20-41  
 [6] R6_2.5.1          rlang_1.1.1       fastmap_1.1.1     tools_4.0.2       pkgbuild_1.4.2   
[11] grid_4.0.2        xfun_0.16         cli_3.6.1         htmltools_0.5.6.1 remotes_2.4.2.1  
[16] yaml_2.2.1        digest_0.6.33     rprojroot_2.0.3   lifecycle_1.0.3   crayon_1.3.4     
[21] processx_3.8.2    Matrix_1.6-1.1    callr_3.7.3       ps_1.7.5          curl_4.3         
[26] evaluate_0.23     rmarkdown_2.3     compiler_4.0.2    desc_1.4.2        scales_1.2.1     
[31] prettyunits_1.1.1

**Comment by igrabski** [comment]
Hi, these issues are most likely related to the older version of R. Please try with the current version of R, and if you are still having problems, please open a new issue.
